src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -50,14 +50,31 @@ private function convertUnitRefundToLineItem(ShipmentRefund $shipmentRefund): Li
         Assert::isInstanceOf($shipment, AdjustableInterface::class);
         Assert::lessThanEq($shipmentRefund->total(), $shipment->getAdjustmentsTotal());
 
+        $grossValue = $shipmentRefund->total();
+
+        $taxAdjustment = $shipment->getAdjustments(AdjustmentInterface::TAX_ADJUSTMENT)->first();
+        $taxRate = $this->getTaxRateAmount($taxAdjustment) . '%';
+        $taxAmount = (int) ($grossValue * $this->getTaxRateAmount($taxAdjustment));
+        $netValue = $grossValue - $taxAmount;
+
         return new LineItem(
             $shippingAdjustment->getLabel(),
             1,
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            $shipmentRefund->total(),
-            0
+            $netValue,
+            $grossValue,
+            $netValue,
+            $grossValue,
+            $taxAmount,
+            $taxRate
         );
     }
+
+    private function getTaxRateAmount(AdjustmentInterface $adjustment): ?float
```

Comment:
The calculation itself could be extracted as well.

Also, taking into account one of the usages of this function:
```php
$taxAmount = (int) ($grossValue * $this->getTaxRateAmount($taxAdjustment));
```
I'm not sure, we should return `null` from this function. In the example above it will be casted to 0. Won't it be better to always return float, sometimes it will be just 0? On the other hand, do we want to inform users of this function that tax rate was not found?